### PR TITLE
feat: add backups server (hal)

### PR DESCRIPTION
**Pull request by nikhiljha**
This is an exact 1:1 copy of the scripts from the original puppet but I replaced the shebang and added relevant packages to systemPackages. Currently untested, not sure if this works.

### CODE REVIEW

File: profiles/backups/assets/rsnapshot-zfs-git.conf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#                                               #
+#################################################
+
+config_version	1.2
+
+cmd_cp	/bin/cp
+cmd_rm	/bin/rm
+cmd_rsync	/usr/local/bin/rsync-no-vanished
+cmd_ssh	/usr/bin/ssh
+cmd_logger	/usr/bin/logger
```
Comment (oliver-ni) [inline profiles/backups/assets/rsnapshot-zfs-git.conf:21]: Might need to change paths? are these remote

File: profiles/backups/assets/rsnapshot-zfs-mysql.conf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#                                               #
+#################################################
+
+config_version	1.2
+
+cmd_cp	/bin/cp
+cmd_rm	/bin/rm
+cmd_rsync	/usr/local/bin/rsync-no-vanished
+cmd_ssh	/usr/bin/ssh
+cmd_logger	/usr/bin/logger
```
Comment (oliver-ni) [inline profiles/backups/assets/rsnapshot-zfs-mysql.conf:21]: Maybe paths again

Comment (oliver-ni) [reply profiles/backups/assets/rsnapshot-zfs-mysql.conf:21]: & on all the other files

File: profiles/backups/assets/rsnapshot.conf [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# scripts
+backup_script	/opt/share/backups/backup-mysql	mysql/
+backup_script	/opt/share/backups/backup-git	git/
+backup_script	/opt/share/backups/backup-pgsql	pgsql/
+
+# remote servers
+backup	ocfbackups@hal:/etc/libvirt/qemu/	servers/vm_xml/hal/
+backup	ocfbackups@pandemic:/etc/libvirt/qemu/	servers/vm_xml/pandemic/
+backup	ocfbackups@riptide:/etc/libvirt/qemu/	servers/vm_xml/riptide/
+backup	ocfbackups@scurvy:/etc/libvirt/qemu/	servers/vm_xml/scurvy/
```
Comment (oliver-ni) [inline profiles/backups/assets/rsnapshot.conf:58]: Might need auth, can copy from desktop repo (before merge)

https://github.com/oliver-ni/ocf-nix-desktops/blob/main/modules/ocf/auth.nix

File: profiles/backups/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  services.cron.enable = true;
+  services.cron.systemCronJobs = [
+    "00 03 * * * /usr/local/sbin/backup-zfs.sh | tee -a /var/log/zfs-backup.log"
+    "0 0 * * * chronic /opt/share/backups/prune-old-backups"
+  ];
+
+  environment.etc = {
+    # rsync backup scripts
+    "ocf_backup/backup-git".source = ./assets/backup-git;
```
Comment (oliver-ni) [inline profiles/backups/default.nix:23]: can you just put these all at once lol

**Review by oliver-ni — CHANGES_REQUESTED** [review]

File: profiles/backups/default.nix (hunk 4 above)
Comment (oliver-ni) [reply profiles/backups/default.nix:23]: you can put source as dir

File: profiles/backups/assets/rsnapshot-zfs-git.conf (hunk 1 above)
Comment (nikhiljha) [reply profiles/backups/assets/rsnapshot-zfs-git.conf:21]: using envfs for now, just so this is 1:1 and easy

File: profiles/backups/assets/check-rsnapshot-backups
Comment (singingtelegram) [inline profiles/backups/assets/check-rsnapshot-backups:1]: this isn't needed, old setup

File: profiles/backups/assets/rsnapshot.conf
Comment (singingtelegram) [inline profiles/backups/assets/rsnapshot.conf:1]: this isn't needed

**Review by singingtelegram — COMMENTED** [review]
see comment. 2 old files, otherwise good

**Pull request closed without merging**